core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -1420,16 +1429,17 @@ class SpendReportingServiceSpec extends AnyFlatSpecLike with Matchers with Mocki
       )
     )
 
-    val exceptionFuture = recoverToExceptionIf[RawlsExceptionWithErrorReport] {
-      service.getSpendForAllWorkspaces(from, to, 100, 0)
-    }
-    exceptionFuture.map { e =>
-      e.errorReport.statusCode shouldBe Option(StatusCodes.InternalServerError)
-      e.errorReport.message.contains("no workspaces") shouldBe true
-    }
+    val exception = Await.result(recoverToExceptionIf[RawlsExceptionWithErrorReport] {
+                                   service.getSpendForAllWorkspaces(from, to, 100, 0)
+                                 },
+                                 Duration.Inf
+    )
+
+    exception.errorReport.statusCode shouldBe Option(StatusCodes.NotFound)
+    exception.errorReport.message.contains("No workspaces eligible for spend report") shouldBe true
```

Comment:
this unit test, and the `handle errors from bigquery gracefully` test just below, were always passing; in effect, they weren't running. The tests assert on a `Future` but didn't wait for that `Future` to complete. So, the test finished - successfully - while the Future continued to run in the background. I've added the proper `Await`s to them … which also required fixing them up a bit to pass.